packages/reqresp/src/protocols/Metadata.ts

```diff
@@ -1,5 +1,15 @@
 import {allForks, ssz} from "@lodestar/types";
 import {ContextBytesType, Encoding, ProtocolDefinitionGenerator} from "../types.js";
+import {seconds} from "./utils.js";
+
+export const metadataInboundRateLimit = {
+  /**
+   * One peer does not request a lot of metadata
+   * For total we multiply with `defaultNetworkOptions.maxPeers`
```

Comment:
> For total we multiply with `defaultNetworkOptions.maxPeers`

No, you are just multiplying by 55 which may or may not be maxPeers. Other consumers may use different maxPeers thus causing un-wanted rate-limits.

By not setting any total limit at all, you are already limiting the total to `2 * maxPeers`, since otherwise peers will be down-scored. The only point of using total over byPeer is to allow higher usage by some peers but prevent an overall overload if byPeer.quota * maxPeers > what we can manage. For any protocol that you are setting total = byPeer * maxPeers, just remove the total limit.

Reply:
Yes `maxPeers` can be changed runtime. That's why I mentioned `defaultNetworkOptions.maxPeers` and not `network.maxPeers`. If user change that option, he can also change limits by `network.rateLimitMultiplier`.

We don't know how many maximum inbound peers are allowed on the node, so we assume and initialize with that the default value is, which is 55. 

I understand the point of having implicit `maxPeers` limit, but since `total` quota was not set as optional so had to initialize it's value.

Reply:
total and byPeer should be optional params